src/server.ts

```diff
@@ -248,8 +261,19 @@ export class ClinicalTrialMatchingService {
     return new Promise((resolve, reject) => {
       const port = this.port;
       const host = this.host;
-      // It's unclear if we can pass undefined to listen
-      const server = host ? this.app.listen(port, host) : this.app.listen(port);
+      // If within IISNode, the port will be a UNC path. Ignore configuration
+      // and use the environment variable instead.
+      const iisnodePort =
+        this.ignoreEnvironment == false && process.env['IISNODE_VERSION'] ? process.env['PORT'] : undefined;
```

Comment:
I would probably change this to:

`!this.ignoreEnvironment && process.env['IISNODE_VERSION'] ? process.env['PORT'] : undefined`